src/lexer.jl

```diff
@@ -40,10 +40,18 @@ mutable struct ParseState
     nws::RawToken
     nnws::RawToken
     closer::Closer
+    lastbyte::Int
     errored::Bool
 end
 function ParseState(str::Union{IO,String})
-    ps = ParseState(tokenize(str, RawToken), false, RawToken(), RawToken(), RawToken(), RawToken(), RawToken(), RawToken(), RawToken(), RawToken(), Closer(), false)
+    lastbyte = if str isa String
+        sizeof(str)
+    elseif str isa IOBuffer
+        str.size
+    else
+        Inf
```

Comment:
Probably best to error here, no?

Reply:
I think I'll just restrict the signature (same effect)